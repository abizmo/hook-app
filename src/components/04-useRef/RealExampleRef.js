import React, { useState } from 'react'
import MultiplesHooks from '../03-examples/MultiplesHooks'

const RealExampleRef = () => {
  const [show, setShow] = useState(false)

  return (
    <div className="div ref-container">
      <h3>RealExampleRef</h3>
      <hr />
      { show && <MultiplesHooks /> }
      <button
        className="btn btn-primary"
        onClick={() => setShow(!show)}
      >Show/Hide</button>
    </div>
  )
}

export default RealExampleRef
