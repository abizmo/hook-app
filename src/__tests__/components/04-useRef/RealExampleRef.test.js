import { shallow } from 'enzyme'
import RealExampleRef from "../../../components/04-useRef/RealExampleRef"

describe('RealExampleRef tests', () => {
  const wrapper = shallow(<RealExampleRef />)

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MultiplesHooks').exists()).toBeFalsy()
  })
  
  test('should show MultiplesHooks when click in Show/Hide', () => {
    wrapper.find('button').simulate('click')

    expect(wrapper.find('MultiplesHooks').exists()).toBeTruthy()
  })
})
