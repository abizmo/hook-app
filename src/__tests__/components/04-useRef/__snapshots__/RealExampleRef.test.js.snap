// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef tests should render and match snapshot 1`] = `
<div
  className="div ref-container"
>
  <h3>
    RealExampleRef
  </h3>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
